test/schedule/swing_test.exs

```diff
@@ -333,41 +322,141 @@ defmodule Schedule.SwingTest do
       }
 
       trips_by_id = %{
-        "5678" =>
+        trip_id_after_swing =>
           build(
             :trip,
-            id: "5678",
+            id: trip_id_after_swing,
             block_id: "A12-34",
-            route_id: "12",
+            route_id: route_id_after_swing,
             service_id: "b",
-            headsign: "Somewhere Else",
-            direction_id: 1,
-            route_pattern_id: "12-1",
-            shape_id: nil,
-            schedule_id: "a",
-            stop_times: [],
-            run_id: "123-789",
-            start_time: 200,
-            end_time: 300,
-            start_place: "place2",
-            end_place: "place1"
+            run_id: run_id_after_swing
+          ),
+        trip_id_during_swing =>
+          build(
+            :trip,
+            id: trip_id_during_swing,
+            block_id: "A12-34",
+            route_id: route_id_during_swing,
+            service_id: "b",
+            run_id: run_id_during_swing
           )
       }
 
-      assert Swing.from_blocks(blocks, trips_by_id) == %{
-               {"b", "12"} => [
+      key = {"b", route_id_during_swing}
+
+      assert %{
+               ^key => [
                  %Swing{
                    block_id: "A12-34",
-                   from_route_id: "12",
-                   from_run_id: "123-456",
-                   from_trip_id: "1234",
-                   to_route_id: "12",
-                   to_run_id: "123-789",
-                   to_trip_id: "5678",
-                   time: 150
+                   from_route_id: ^route_id_during_swing,
+                   from_run_id: ^run_id_during_swing,
+                   from_trip_id: ^trip_id_during_swing,
+                   time: 150,
+                   to_route_id: ^route_id_during_swing,
+                   to_trip_id: ^trip_id_during_swing,
+                   to_run_id: ^run_id_after_swing
                  }
                ]
-             }
+             } = Swing.from_blocks(blocks, trips_by_id)
+    end
+
+    @tag :skip
```

Comment:
**question:** is this intentional? Looks like it may have inadvertently been left in.

Reply:
that test is failing due to the data edge case, but with your other comments we're able to enable it.